Feature-query-and-update-animals-table

maximoortelli opened the pull request:
> ### **FEATURE-QUERY-AND-UPDATE-ANIMALS-TABLES:**

----------------------------------------------------------------------------------------------------------------------------------

_This branch showcases proficiency in crafting SQL queries that yield insightful insights. The queries range from straightforward computations, such as calculating total animal counts and average weights, to more intricate analyses, such as escape patterns based on neutering status and statistical inferences regarding escape attempts across specific birthdate ranges._

----------------------------------------------------------------------------------------------------------------------------------

> ### **DESCRIPTION:**

----------------------------------------------------------------------------------------------------------------------------------

 

- [x] Integrate the species column into the schema of the animal's table, enhancing its data structure.
- [x] Populate the database with a diverse range of animal information, encompassing names, birthdates, weights, 
       neutering status, and escape attempts.
- [x] Transform the species column to unspecified, validating the effectiveness of the alterations.
- [x] Refine species categorization by updating entries with names ending in mon to digimon.
- [x] Rectify instances of undefined species by assigning them the label pokemon.
- [x] Ensure accuracy by verifying changes, committing them, and confirming their continued persistence post-commit.
- [x] Ascertain the total count of animals present in the dataset.
- [x] Identify and isolate animals that have never attempted escape, showcasing an analytical approach.
- [x] Employ statistical methodologies to calculate the average weight of the animal population
- [x] Conduct an insightful analysis of escape behaviors, distinguishing between neutered and non-neutered animals.
- [x] Employ complex aggregations to extract the minimum and maximum weights for each distinct animal type.
- [x] Employ a nuanced query to determine the average number of escape attempts for animals born between the years 1990 and 2000.
- [x] Address schema enhancement requirements by adding a new column named species of string data type to the animal's table.
- [x] Execute data insertion processes to populate the newly introduced species column with relevant information.
- [x] As the requirements said, I add the screenshots of the result of my work.

----------------------------------------------------------------------------------------------------------------------------------

> ### **SCREENSHOTS:**

----------------------------------------------------------------------------------------------------------------------------------

![1](https://github.com/maximoortelli/Vet-clinic-database/assets/118080546/fcb55599-d22b-4e7c-8aca-aabe3e775ba2)
![2](https://github.com/maximoortelli/Vet-clinic-database/assets/118080546/a143e419-b5cf-446e-b42c-bbd32fbabd7f)
![3](https://github.com/maximoortelli/Vet-clinic-database/assets/118080546/11236f7f-c026-4f99-ba56-5f086c832391)
![4](https://github.com/maximoortelli/Vet-clinic-database/assets/118080546/4ecc343f-8495-42c7-888d-01b1ffaf7a5c)
![5](https://github.com/maximoortelli/Vet-clinic-database/assets/118080546/ddd42805-a909-4b7d-9f1c-762c4280c45d)
![6](https://github.com/maximoortelli/Vet-clinic-database/assets/118080546/e2f105ba-6a9f-42a3-b465-5f23ec173dd0)
![7](https://github.com/maximoortelli/Vet-clinic-database/assets/118080546/a77095e2-2c49-48bc-9444-beec80311787)
![8](https://github.com/maximoortelli/Vet-clinic-database/assets/118080546/9613d00c-0b57-467c-9949-a58a7819a003)
![9](https://github.com/maximoortelli/Vet-clinic-database/assets/118080546/1c088efe-037f-43dc-8744-f64fa2e92b01)
![10](https://github.com/maximoortelli/Vet-clinic-database/assets/118080546/2a0694a9-58c4-4a42-95f0-477b275b1424)
![11](https://github.com/maximoortelli/Vet-clinic-database/assets/118080546/dc0aebc1-3f5a-4d8d-8b80-2a54f0246639)
![12](https://github.com/maximoortelli/Vet-clinic-database/assets/118080546/74129ed9-777d-4a6d-a053-79ffa27230e0)
![13](https://github.com/maximoortelli/Vet-clinic-database/assets/118080546/e7df8b11-8fc7-4690-b3bc-3ae621967183)
![14](https://github.com/maximoortelli/Vet-clinic-database/assets/118080546/6292ba17-e3f0-4baa-b88b-f2794846b2d3)
![15](https://github.com/maximoortelli/Vet-clinic-database/assets/118080546/229384ae-ac29-469d-93e1-fc2ce224857e)
![16](https://github.com/maximoortelli/Vet-clinic-database/assets/118080546/220e0e38-56a2-4be9-ac56-76fb7a990d54)
![17](https://github.com/maximoortelli/Vet-clinic-database/assets/118080546/82ac6198-91fd-4bc7-ad2a-6f9c08f0ac5a)
![18](https://github.com/maximoortelli/Vet-clinic-database/assets/118080546/e58ad3bd-519d-4552-81a3-81d2431b60c4)
![19](https://github.com/maximoortelli/Vet-clinic-database/assets/118080546/7926f42f-7e5b-47cb-ad66-d8e3490d2b18)
![20](https://github.com/maximoortelli/Vet-clinic-database/assets/118080546/96595e34-2629-4e12-9f31-58a53d2839b2)
![21](https://github.com/maximoortelli/Vet-clinic-database/assets/118080546/08494c1a-d927-4d48-af77-dbdc7f74171f)
![22](https://github.com/maximoortelli/Vet-clinic-database/assets/118080546/37bde660-b3c0-4efe-9d77-20db6fd43ec2)
![23](https://github.com/maximoortelli/Vet-clinic-database/assets/118080546/d8951f43-513c-4f5b-97af-734b31eff0de)
![24](https://github.com/maximoortelli/Vet-clinic-database/assets/118080546/46f63c1c-ccf1-4697-b10a-9b98d55998c3)
![25](https://github.com/maximoortelli/Vet-clinic-database/assets/118080546/c7fc6fe3-7e91-48f3-b1a1-5fb5e5eea326)
![26](https://github.com/maximoortelli/Vet-clinic-database/assets/118080546/2a9797a4-f54a-43cd-b186-a052ea58c3b6)
![27](https://github.com/maximoortelli/Vet-clinic-database/assets/118080546/ca095d64-e85c-4a63-97a2-73a7bd8fb088)
![28](https://github.com/maximoortelli/Vet-clinic-database/assets/118080546/3973776a-07a1-4ff2-9c2b-aa5869dbe137)
![29](https://github.com/maximoortelli/Vet-clinic-database/assets/118080546/f671949f-ce6b-4eb5-8c58-ba8ef96dcd9e)
![30](https://github.com/maximoortelli/Vet-clinic-database/assets/118080546/9fbdfa5c-57bf-402e-b4bc-ad818304a896)
![31](https://github.com/maximoortelli/Vet-clinic-database/assets/118080546/b7d6cf9e-0118-4b92-a242-2225ea38327a)

maximoortelli committed:
Update all files

Update all files

LionRouge1 commented on data.sql:
```diff
@@ -11,3 +11,24 @@ VALUES ('Pikachu', '2021-01-07', 1, false, 15.04);
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
+VALUES ('Angemon', '2005-06-12', 1, true, -45.0);
+
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
+VALUES ('Boarmon', '2005-06-07', 7, true, 20.4);
+
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
+VALUES ('Blossom', '1998-10-13', 3, true, 17.0);
+
```
- [x] Nice work inserting the data (animals) in the animals' table :clap: However the code for adding the animal named **Ditto** is not present in your code. Kindly make sure all the animals are inserted into the table as requested in the project requirement.
![Screenshot from 2023-10-26 22-56-01](https://github.com/maximoortelli/Vet-clinic-database/assets/91142432/cde4eb9d-14e3-4259-8bfc-9f2b38e7088c)

maximoortelli replied:
Done, thank you for the review 👍😃

LionRouge1 commented on queries.sql:
```diff
@@ -22,3 +22,93 @@ WHERE name <> 'Gabumon';
+
+-- First transaction
+
+BEGIN;
+
+UPDATE animals
+SET species = 'unspecified';
+
+SELECT * FROM animals;
+
+ROLLBACK;
```
- [x] According to the project requirement, we should verify that the change was made and rollbacked. Therefore I suggest also checking  if the change was reversed by adding `SELECT * FROM animals;` after the **ROLLBACK** keyword

maximoortelli replied:
Done, thank you for the review 👍😃

LionRouge1 reviewed (changes requested):
Hi @maximoortelli, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

![almost](https://media.giphy.com/media/KyBesf8kNT6s88SEgb/giphy.gif)

## Highlights
- Use transactions to update and delete records :heavy_check_mark: 
- Use aggregate functions and GROUP BY to answer analytical questions :heavy_check_mark: 
- Insert all the animals listed in the project requirement into their tables :x: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

maximoortelli committed:
Update data.sql

I add INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
VALUES ('Ditto', '2022-05-14', 4, true, 22);

maximoortelli committed:
Update queries.sql

I have checked if the change was reverted by adding SELECT * FROM animals; after the ROLLBACK keyword

zeff96 commented on queries.sql:
```diff
@@ -22,3 +22,93 @@ WHERE name <> 'Gabumon';
+BEGIN;
+
+UPDATE animals
+SET species = 'unspecified';
```
- [x] 👋 Hello! Fantastic job on the implementation of the query transaction! 🚀 Please make sure to double-check the modifications, as the previous reviewer suggested 🕵️, before considering any rollback 🔄. This extra step helps ensure that any changes made are thoroughly validated and in line with our requirements 📝. Thank you for your attention to detail! 🙌😊
```suggestion
SELECT species from animals; -- verify that change was made
```

maximoortelli replied:
Done, thanks for you review 👍😃

zeff96 commented on data.sql:
```diff
@@ -11,3 +11,27 @@ VALUES ('Pikachu', '2021-01-07', 1, false, 15.04);
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
+VALUES ('Ditto', '2022-05-14', 4, true, 22);
+
+SELECT * FROM animals;
```
- [ OPTIONAL ] 👋 Hey there! Let's make sure that we've got all our queries neatly organized in the "queries.sql" file 📁. This will help keep our project well-structured and make it easier for everyone to locate and manage the queries efficiently. Thank you for your attention to this important detail! 🙌😊

maximoortelli replied:
Done, thanks for you review 👍😃

zeff96 commented on queries.sql:
```diff
@@ -22,3 +22,93 @@ WHERE name <> 'Gabumon';
+FROM animals;
+
+-- Who escapes the most, neutered or not neutered animals?
+SELECT neutered, SUM(escape_attempts) AS total_escape_attempts
```
- [ OPTIONAL ] 👋 Hey! To address the question about which animals, neutered or not, have escaped the most, it might be more informative to return the specific animal with the highest number of escape attempts 📈, rather than just the sum of escape attempts. 🐾 This way, we can pinpoint the individual animal that has had the most escapes, offering a more precise and insightful answer to the query. 🐕🦴🐈🏃‍♂️ Thank you for considering this improvement! 🙌😊
- 
```suggestion
 SELECT neutered, MAX(escape_attempts) FROM animals
  GROUP BY neutered;
```

maximoortelli replied:
Done, thanks for you review 👍😃

zeff96 reviewed (changes requested):
### Status: CHANGES REQUIRED ♻️
Hi @maximoortelli , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

![almost](https://media0.giphy.com/media/t8YPf3ACKeouLzEIvr/200w.webp?cid=ecf05e47ridpj5qdjw5d89yscw4j96jhddchsboqdc6tgqov&ep=v1_gifs_search&rid=200w.webp&ct=g)

 ## Highlights
✔️ Correct GitFlow
✔️ Descriptive pull request

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

maximoortelli committed:
Update files

Update files

ShadyShawkat reviewed (approved):
## Hi @maximoortelli, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 
- Your readme looks professional ✔️ 

![Image](https://media.giphy.com/media/jJQC2puVZpTMO4vUs0/giphy.gif)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
